R/package.R

```diff
@@ -50,6 +50,10 @@
 argufy <- function(fun, ...) {
   if (!is.function(fun)) stop("'fun' must be a function")
 
+  # these statements are needed to get S4 functions to work properly
+  was_s4 <- isS4(fun)
+  old_attributes <- attributes(fun)
+
```

Comment:
That is true.  In some respects the `srcref` version is more useful interactively, because the function definition is not cluttered by the additional checking code (which will become a larger problem when inlining functions (https://github.com/gaborcsardi/argufy/issues/6)).  On the other hand the printed function is not exactly what is called by R, which may make things confusing.

It is possible to view the real source with `print(fun, useSource = FALSE)` even with the copied srcref.

The only downside to not having a `srcref` at all is that you lose comments and formatting, because R then simply prints the parsed expressions, it shouldn't actually cause any errors anywhere though.

I personally would lean towards leaving the srcref copied as it is, but if you feel it is confusing simply dropping that attribute shouldn't cause any problems.